server/message_hooks.go

```diff
@@ -14,13 +14,28 @@ import (
 	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams"
 	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams/clientmodels"
 	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
-	"github.com/mattermost/mattermost-server/v6/model"
-	"github.com/mattermost/mattermost-server/v6/plugin"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
 	"github.com/microsoftgraph/msgraph-sdk-go/models"
 	"github.com/pkg/errors"
 	"gitlab.com/golang-commonmark/markdown"
 )
 
+func (p *Plugin) UserWillLogIn(_ *plugin.Context, user *model.User) string {
+	p.API.LogDebug("Promoting synthetic user", "UserID", user.Id)
```

Comment:
Also, we should only log this if we actually promote the user -- otherwise this log gets emitted for every user that logs in.